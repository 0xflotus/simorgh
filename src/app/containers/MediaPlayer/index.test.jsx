import { shouldMatchSnapshot, isNull } from '@bbc/psammead-test-helpers';
import {
  VideoCanonicalWithPlaceholder,
  VideoCanonicalNoPlaceholder,
  VideoAmp,
  VideoCanonicalNoVersionId,
  VideoCanonicalToggledOff,
  VideoCanonicalWithCaption,
  VideoAmpWithCaption,
} from './fixtureData';

describe('MediaPlayer', () => {
  shouldMatchSnapshot(
    'Calls the canonical placeholder when platform is canonical and showPlaceholder is true',
    VideoCanonicalWithPlaceholder,
  );

  shouldMatchSnapshot(
    'Does not Call the canonical placeholder when platform is canonical but showPlaceholder is false',
    VideoCanonicalNoPlaceholder,
  );

  shouldMatchSnapshot('Calls the AMP player when platform is AMP', VideoAmp);

  shouldMatchSnapshot(
    'Calls the canonical player with a caption',
    VideoCanonicalWithCaption,
  );

  shouldMatchSnapshot(
    'Calls the AMP player with a caption',
    VideoAmpWithCaption,
  );

  describe('Fails and returns early when', () => {
    isNull('there is no versionId', VideoCanonicalNoVersionId);
    isNull('component is toggled off', VideoCanonicalToggledOff);
  });
});
