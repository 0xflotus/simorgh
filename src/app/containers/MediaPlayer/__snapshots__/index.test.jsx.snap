// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPlayer Calls the canonical placeholder when platform is canonical and showPlaceholder is true 1`] = `
.c2 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c1 {
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c0 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
  </head>
  <body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <img
            alt="Image Alt"
            class="c2"
            src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
          />
        </div>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Does not Call the canonical placeholder when platform is canonical but showPlaceholder is false 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
  </head>
  <body>
    .c1 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border: 0;
  overflow: hidden;
}

.c0 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

<div>
      <div
        class="c0"
      >
        <iframe
          allow="autoplay; fullscreen"
          allowfullscreen=""
          class="c1"
          src="https://www.test.bbc.com/ws/av-embeds/articles/c123456789o/p01k6msp/en-GB"
          title="Media player"
        />
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Renders the AMP player when platform is AMP 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    .c0 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

<div>
      <div
        class="c0"
      >
        <amp-iframe
          allowfullscreen="allowfullscreen"
          frameborder="0"
          layout="fill"
          sandbox="allow-scripts allow-same-origin"
          src="https://www.test.bbc.com/ws/av-embeds/articles/c123456789o/p01k6msp/en-GB/amp"
          title="Media player"
        >
          <amp-img
            layout="fill"
            placeholder="true"
            src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
          />
        </amp-iframe>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Renders the AMP player with a caption 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    .c0 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c2 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c1 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: italic;
  color: #6E6E73;
  padding: 0.5rem 0.5rem 0;
  width: 100%;
}

.c1 > p {
  padding-bottom: 1.5rem;
  margin: 0;
}

.c1 > p:last-child {
  padding-bottom: 0;
}

.c1 i {
  font-style: normal;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c1 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c1 {
    padding: 0.5rem 1rem 0;
  }
}

@media (min-width:63rem) {
  .c1 {
    padding: 0.5rem 0 0;
  }
}

<div>
      <div
        class="c0"
      >
        <amp-iframe
          allowfullscreen="allowfullscreen"
          frameborder="0"
          layout="fill"
          sandbox="allow-scripts allow-same-origin"
          src="https://www.test.bbc.com/ws/av-embeds/articles/c123456789o/p01k6msp/en-GB/amp"
          title="Media player"
        >
          <amp-img
            layout="fill"
            placeholder="true"
            src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
          />
        </amp-iframe>
      </div>
      <figcaption
        class="c1"
      >
        <span
          class="c2"
        >
          Video caption, 
        </span>
        <p>
          Media Player With Caption
        </p>
      </figcaption>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer Renders the canonical player with a caption 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"video":{"@list":[{"@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":191,"thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":1540218932000}]}}
    </script>
  </head>
  <body>
    .c2 {
  display: block;
  width: 100%;
  visibility: visible;
}

.c1 {
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.c0 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c4 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.c3 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: italic;
  color: #6E6E73;
  padding: 0.5rem 0.5rem 0;
  width: 100%;
}

.c3 > p {
  padding-bottom: 1.5rem;
  margin: 0;
}

.c3 > p:last-child {
  padding-bottom: 0;
}

.c3 i {
  font-style: normal;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c3 {
    padding: 0.5rem 1rem 0;
  }
}

@media (min-width:63rem) {
  .c3 {
    padding: 0.5rem 0 0;
  }
}

<div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <img
            alt="Image Alt"
            class="c2"
            src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
          />
        </div>
      </div>
      <figcaption
        class="c3"
      >
        <span
          class="c4"
        >
          Video caption, 
        </span>
        <p>
          Media Player With Caption
        </p>
      </figcaption>
    </div>
  </body>
</html>
`;
